rviz_rendering/src/rviz_rendering/objects/point_cloud.cpp

```diff
@@ -587,6 +587,10 @@ PointCloud::addPointToHardwareBuffer(
   auto num_vertices = internals.rend->getBuffer()->getNumVertices();
   auto vertex_size =
     internals.rend->getRenderOperation()->vertexData->vertexDeclaration->getVertexSize(0);
+  // When the package is build in Release mode these variables are unused
+  (void) num_vertices;
+  (void) vertex_size;
```

Comment:
The problem with this is that while it hides the warning, it also does unnecessary work when the `assert` is compiled out.  The optimizer in the compiler will likely take care of that, but I'd rather that we explicitly do that in all cases.

Looking at the `assert` again, I think it is actually bogus.  `float_buffer` is being incremented on every loop, so it will *always* be `<=` itself plus the `num_vertices * vertex_size`.  What I think that really wants to compare is whether `float_buffer` is `<=` the original buffer, `internals.float_buffer`.

And further, I think it is probably sufficient to do that at the end of the loop once, which will make it even faster.  So I'm thinking that we move the 4 lines above, plus the assert, to right after the end of the loop (before the `internals.float_buffer = float_buffer`, and make it look something like this:

```
#ifndef NDEBUG
  size_t num_vertices = internals.rend->getBuffer()->getNumVertices();
  size_t vertex_size =
    internals.rend->getRenderOperation()->vertexData->vertexDeclaration->getVertexSize(0);
  assert(
    reinterpret_cast<uint8_t *>(float_buffer) <=
    reinterpret_cast(internals.float_buffer) + num_vertices * vertex_size);
#endif
```

What do you think?